### DISCUSSION TITLE
Amiberry and RaspiOS 64 Bit. Some installation questions about autostart, screen resolution and rabbithole.

### DISCUSSION THREAD

**Discussion by Retro1968** [BlitterStudio/amiberry#926] [General]
Raspberry Pi 4 running RaspiOS 64 Bit with amiberry 5.0 and AmigaOS 3.2.1
I compile with: make -j4 PLATFORM=rpi4-64-sdl2

Hello Dimitris,
Hello forum members,

I set up amiberry 5.0 under RaspiOS 64 bit and found that the sound output is better there than under Manjaro Linux.

I notice this particularly with the intro of Lotus 1. Under Manjaro Linux I still have a slight croaking while under RaspiOS 64 Bit with the same sound settings it runs completely error-free. I also no longer have occasional sound stutters in videos. But I still have nothing against Manjaro Linux and this is not the reason for my questions.

Rather, I have five other problems with RaspiOS 64 bit and would appreciate any help.

1. How can I change the desktop screen resolution of RaspiOS 64 bit to 800 x 600?

I already put these settings in my config.txt:
hdmi_force_hotplug=1
hdmi_group=2
hdmi_mode=9
hdmi_ignore_edid=0xa5000080

When booting, 800x600 is displayed, but not on the desktop of RaspiOS 64 Bit, although I have set the "Headless Resolution" to 800x600 in the "Raspberry Pi Configuration" GUI. I also set the resolution to 800x600 in the "Screen Layout Editor". But with every restart, however, it resets to 1280x1024 on my 4:3 LCD monitor. I can't get any further.

2. **[SOLVED]** How can I start amiberry automatically? I have amiberry in the "/home/pi/amiberry/" folder.
I tried several options on the internet, but none of them worked successfully. **[SOLVED]**

3. I use rabbitholes under AmigaOS 3.2.1. This also works well under RaspiOS 64 Bit. But The LXTerminal doesn't want to start when I create a rabbithole with "host-run lxterminal". On the other hand, if I start the LXTerminal under AmigaOS 3.2.1 in the shell with the command "host-run lxterminal" it starts without any problems. What am I doing wrong with my rabbithole file? 
Under Manjaro Linux this works: host-run exo-open --launch TerminalEmulator

4. My rabbithole for the file manager doesn't work either.  
Under Manjaro Linux this works: host-run exo-open --launch FileManager

5. My rabbithole for the filezilla doesn't work either. If i start it from the AmigaOS 3.2.1 shell with "host-run filezilla", filezilla runs without problems.

Cheers
Retro1968

**Comment by midwan** [comment]
This doesn't seem like an Amiberry issue, but more related to a discussion - moving it there.

**Comment by Retro1968** [comment]
I have now found the following solution for question 2, so that amiberry is started automatically when RaspiOS 64bit is started and RaspiOS 64bit is automatically shut down when amiberry is closed:

In the /home/pi folder, create the file launch-amiberry: nano /home/pi/launch-amiberry

Paste these lines.

cd /home/pi/amiberry
./amiberry
shutdown now

Save with CTRL+O and exit with CTRL+X.
Assign file permissions: sudo chmod 755 /home/pi/launch-amiberry

Then: sudo nano /etc/xdg/lxsession/LXDE-pi/autostart

and add this as the last entry: @lxterminal -e ./launch-amiberry

Voila! ;-)

**Reply by midwan** [reply]
You don't need the first line in your script, since you're doing a `cd` to an absolute path in the second line anyway. ;)

**Reply by Retro1968** [reply]
> You don't need the first line in your script, since you're doing a `cd` to an absolute path in the second line anyway. ;)

Yes thx, i corrected that.

**Comment by Retro1968** [comment]
I updated my first post. :-)

**Comment by Retro1968** [comment]
Could the rabbit hole problems be a host-run problem?
Since even the Filzilla rabbithole, which was started under Manjaro simply with "host-run filezilla", cannot be started under RaspiOS 64 bit.

The situation is different under Manjaro with "host-run exo-open --launch TerminalEmulator", where "exo-open --launch" is a special Manjaro Linux command that is explained here: https://www.systutorials.com/docs/linux/man/1-exo-open/

I haven't been able to find out if there is a similar command for RaspiOS 64 bit.

Any ideas for a solution?

**Reply by midwan** [reply]
If it runs from the AmigaOS console, using the command you mentioned above, then the problem is not host-run itself. It sounds more like something is wrong with your icon or wherever you're running it from, when it's not the console.

Remember, it needs to be run as an AmigaDOS command, not a Workbench tool. Normally you'd use an icon with the type of Project, and run it with `iconX`. Alternatively make sure it's executed from the shell.

**Reply by Retro1968** [reply]
> If it runs from the AmigaOS console, using the command you mentioned above, then the problem is not host-run itself. It sounds more like something is wrong with your icon or wherever you're running it from, when it's not the console.
> 
> Remember, it needs to be run as an AmigaDOS command, not a Workbench tool. Normally you'd use an icon with the type of Project, and run it with `iconX`. Alternatively make sure it's executed from the shell.

Yes, "host-run filezilla" runs in AmigasOS shell. When I compare the properties of filezilla- and firefox-rabbitholeicons, under the pictogram both say 4096KB and start with "Shell". Both have all 5 flags set except for the archive flag. Firefox will then start normally. Filezilla not. I increased the memory to 8192KB, but it doesn't start even with that.

**Reply by Retro1968** [reply]
> Remember, it needs to be run as an AmigaDOS command, not a Workbench tool. Normally you'd use an icon with the type of Project, and run it with `iconX`. 

This do not work on my Filezilla rabbithole... 
Would it be possible to send you my rabbitholes to check for errors on RaspiOS 64 bit?
